cc/tf_bt_utils.cc

```diff
@@ -118,23 +119,55 @@ void WriteGameExamples(const std::string& gcp_project_name,
             << examples.size() << std::endl;
 }
 
+void WriteEvalRecord(const std::string& gcp_project_name,
+                     const std::string& instance_name,
+                     const std::string& table_name,
+                     const MctsPlayer& player,
+                     const std::string& black_player_name,
+                     const std::string& white_player_name,
+                     const std::string& sgf_name) {
+  using namespace google::cloud::bigtable;
+  Table table(CreateDefaultDataClient(gcp_project_name, instance_name,
+                                      ClientOptions()),
+              table_name);
+
+  // Retrieve the game counter from the Bigtable and increment it atomically.
+  uint64_t game_counter = IncrementGameCounter(
+      gcp_project_name, instance_name, table_name, "eval_game_counter", 1);
+
+  auto row_name = absl::StrFormat(kEvalGameRowFormat, game_counter);
+  google::cloud::bigtable::SingleRowMutation row_mutation(
+      row_name, SetCell("metadata", "black", black_player_name),
+      SetCell("metadata", "white", white_player_name),
+      SetCell("metadata", "black_won", absl::StrCat(player.result() > 0)),
+      SetCell("metadata", "white_won", absl::StrCat(player.result() < 0)),
+      SetCell("metadata", "result", player.result_string()),
+      SetCell("metadata", "length", absl::StrCat(player.history().size())),
+      SetCell("metadata", "sgf", sgf_name));
+
+  table.Apply(std::move(row_mutation));
+  std::cerr << "Bigtable eval row written to " << row_name << std::endl;
+}
+
 uint64_t IncrementGameCounter(const std::string& gcp_project_name,
                               const std::string& instance_name,
-                              const std::string& table_name, size_t delta) {
+                              const std::string& table_name,
+                              const std::string& counter_name,
+                              size_t delta) {
+  using namespace google::cloud::bigtable;
   Table table(CreateDefaultDataClient(gcp_project_name, instance_name,
-                                      google::cloud::bigtable::ClientOptions()),
+                                      ClientOptions()),
               table_name);
-  using namespace google::cloud::bigtable;
   auto rule =
-      ReadModifyWriteRule::IncrementAmount("metadata", "game_counter", delta);
+      ReadModifyWriteRule::IncrementAmount("metadata", counter_name, delta);
   auto row = table.ReadModifyWriteRow("table_state", rule);
   for (auto const& cell : row.cells()) {
     if (cell.family_name() == "metadata" &&
-        cell.column_qualifier() == "game_counter") {
+        cell.column_qualifier() == counter_name) {
       return cell.value_as<bigendian64_t>().get();
     }
   }
-  MG_FATAL() << "Failed to increment table_state=metadata:game_counter";
+  MG_FATAL() << "Failed to increment table_state=metadata:" << counter_name;
```

Comment:
@gitosaurus Couple of questions here:
1. Why do we iterate row.cells()? Should we ever expect more than one value?
1. Is there a way to set a uint value from the CLI?
    * something like `cbt -project seth-minigo -instance minigo-instance set games table_state metadata:game_counter='\x00\x00\x00\x00\x00\x00\x00\f'`
1. The only downside to using this method is that I have to create the Table (line 158) twice for each table write, can you confirm that table object is "cheap" to create.

Reply:
1. The reason I iterate is because it does return a collection, and I wanted to be robust against any future changes to our code or the client code where we might get back more than one cell from the `ReadModifyWriteRow`. At present, the [documentation](https://googlecloudplatform.github.io/google-cloud-cpp/0.2.0/bigtable/classgoogle_1_1cloud_1_1bigtable_1_1v0_1_1Table.html#af23d1796d60442eac811eac99808ad9f) states simply that "the row" is returned.
2. I don't know how to set a uint64 value from the `cbt` CLI (which is one reason to use them sparingly). From C++ and Python, it is straightforward.
3. Construction of the table object (which is just a handle to the client) is quite cheap. Profiling on Python, table creation is 2 microseconds, compared to 6 milliseconds for a row commit.

Reply:
1. Thank you fro the explanation.
2. Ack, I ended up using python which was easy.
3. Thank you for measuring / knowing